documentation/environment-variables.md

```diff
@@ -0,0 +1,26 @@
+# Environment Variables
+
+Environment variables are created and used in many places throughout the repo.  This document covers the use of environment variables in a development environment.  Environment management in production is out of scope.
+
+## Laravel
+
+The Laravel projects (api and auth) read environment variables from the `.env` files in the root of their project directories.  These files are not checked into version control.  When running the setup scripts a new one is copied from the `.env.example` template and then customized.  In a development environment these files are reread on every new PHP request.  If needed a configuration reset can be forced with `php artisan config:clear`.
+
+## Frontend
+
+The frontend React projects uses environment variables in two distinct ways: **build-time** variables and **run-time** variables.
+
+### Build-Time Variables
+
+The frontend subprojects (admin, talentsearch, etc) have `.env` files that are copied from `.env.example` template files like the Laravel projects.  When the frontend projects are built using Webpack, variables from the `.env` file are defined in the resulting bundle using DefinePlugin.  These values are "baked into" the bundle any if there are any variable changes the subproject must be rebuilt to reflect them.
+
+### Run-Time Variables
+
+For deployment in production there needs to be a way to change variables in the program without rebuilding.  This is done using Apache Server Side Includes and the `config.sjs` files.  In a development environment these variables are injected into the docker container from the `/frontend/.apache_env` file.  To update these variables in the container a new container must be built.  For example:
+1) `docker-compose stop php`
+2) `docker-compose rm php`
+3) `docker-compose up -d`
```

Comment:
I believe this works the same: `docker-compose up --detach --build php`

Also, for your consideration: I've seen it advocated to avoid using short arg flags in both documentation and scripts, and prefer long form. The rationale being that the short flags are made for saving keystrokes when people know what they're doing, but documentation and scripts should always assume their intended audience is learners, for whom the long form helps make actions legible. (of course, always exceptions, but for me, blanket rules just feel simpler... rather than asking "am i expecting experts to be reading this?" all the time)

Reply:
Hmm, I just tried the one-liner and it appears to rebuild the image but doesn't recreate a running container.

I agree 100% about using the long-form for documentation and scripts though.  Good catch, I'll update that one.